src/measure_theory/function/jacobian.lean

```diff
@@ -0,0 +1,1274 @@
+  have J : ∀ᶠ a in 𝓝[>] (0 : ℝ≥0∞), μ (f '' s) ≤ m * (μ s + a),
+  { filter_upwards [self_mem_nhds_within],
+    assume a ha,
+    change 0 < a at ha,
+    obtain ⟨t, r, t_count, ts, rpos, st, μt⟩ : ∃ (t : set E) (r : E → ℝ), t.countable ∧ t ⊆ s
+      ∧ (∀ (x : E), x ∈ t → 0 < r x) ∧ (s ⊆ ⋃ (x ∈ t), closed_ball x (r x))
+      ∧ ∑' (x : ↥t), μ (closed_ball ↑x (r ↑x)) ≤ μ s + a :=
+        besicovitch.exists_closed_ball_covering_tsum_measure_le μ ha.ne' (λ x, Ioi 0) s
+        (λ x xs δ δpos, ⟨δ/2, by simp [half_pos δpos, half_lt_self δpos]⟩),
+    haveI : encodable t := t_count.to_encodable,
+    calc μ (f '' s)
+        ≤ μ (⋃ (x : t), f '' (s ∩ closed_ball x (r x))) :
+      begin
+        rw bUnion_eq_Union at st,
+        apply measure_mono,
+        rw [← image_Union, ← inter_Union],
+        exact image_subset _ (subset_inter (subset.refl _) st)
+      end
+    ... ≤ ∑' (x : t), μ (f '' (s ∩ closed_ball x (r x))) : measure_Union_le _
+    ... ≤ ∑' (x : t), m * μ (closed_ball x (r x)) :
+      ennreal.tsum_le_tsum (λ x, I x (r x) (ts x.2) (rpos x x.2).le)
+    ... ≤ m * (μ s + a) :
+      by { rw ennreal.tsum_mul_left, exact ennreal.mul_le_mul le_rfl μt } },
+  -- taking the limit in `a`, one obtains the conclusion
+  have L : tendsto (λ a, (m : ℝ≥0∞) * (μ s + a)) (𝓝[>] 0) (𝓝 (m * (μ s + 0))),
+  { apply tendsto.mono_left _ nhds_within_le_nhds,
+    apply ennreal.tendsto.const_mul (tendsto_const_nhds.add tendsto_id),
+    simp only [ennreal.coe_ne_top, ne.def, or_true, not_false_iff] },
+  rw add_zero at L,
+  exact ge_of_tendsto L J,
+end
+
+/-- Let `f` be a function which is sufficiently close (in the Lipschitz sense) to a given linear
+map `A`. Then it expands the volume of any set by at least `m` for any `m < det A`. -/
+lemma mul_le_add_haar_image_of_lt_det
+  (A : E →L[ℝ] E) {m : ℝ≥0} (hm : (m : ℝ≥0∞) < ennreal.of_real (|A.det|)) :
+  ∀ᶠ δ in 𝓝[>] (0 : ℝ≥0), ∀ (s : set E) (f : E → E) (hf : approximates_linear_on f A s δ),
+  (m : ℝ≥0∞) * μ s ≤ μ (f '' s) :=
+begin
+  apply nhds_within_le_nhds,
+  -- The assumption `hm` implies that `A` is invertible. If `f` is close enough to `A`, it is also
+  -- invertible. One can then pass to the inverses, and deduce the estimate from
+  -- `add_haar_image_le_mul_of_det_lt` applied to `f⁻¹` and `A⁻¹`.
+  -- exclude first the trivial case where `m = 0`.
+  rcases eq_or_lt_of_le (zero_le m) with rfl|mpos,
+  { apply eventually_of_forall,
+    simp only [forall_const, zero_mul, implies_true_iff, zero_le, ennreal.coe_zero] },
+  have hA : A.det ≠ 0,
+  { assume h, simpa only [h, ennreal.not_lt_zero, ennreal.of_real_zero, abs_zero] using hm },
+  -- let `B` be the continuous linear equiv version of `A`.
+  let B := ((A : E →ₗ[ℝ] E).equiv_of_det_ne_zero hA).to_continuous_linear_equiv,
+  have : (B : E →L[ℝ] E) = A,
+  { ext x,
+    simp only [linear_equiv.of_is_unit_det_apply, continuous_linear_equiv.coe_coe,
+      continuous_linear_map.coe_coe, linear_equiv.coe_to_continuous_linear_equiv'] },
```

Comment:
Maybe obtaining that linear equiv is something that we will want to do frequently enough for that have to be a lemma? And should we have a def `linear_map.continuous_linear_equiv_of_det_ne_zero`?

Reply:
Done.